vistopia/visitor.py

```diff
@@ -114,19 +114,64 @@ def save_transcript(self, id: int, episodes: Optional[set] = None):
                     with open(fname, "w") as f:
                         f.write(content)
 
+    def save_transcript_with_single_file(self, id: int,
+                                         episodes: Optional[set] = None,
+                                         single_file_exec_path: str = "",
+                                         cookie_file_path: str = ""):
+        import subprocess
+        from pathlib import Path
+        logger.debug(f"save_transcript_with_single_file id {id}")
+
+        catalog = self.get_catalog(id)
+        show_dir = Path(catalog["title"])
+        show_dir.mkdir(exist_ok=True)
+
+        for part in catalog["catalog"]:
+            for article in part["part"]:
+                if episodes and int(article["sort_number"]) not in episodes:
+                    continue
+
+                fname = show_dir / "{}.html".format(article["title"])
+                if not fname.exists():
+                    command = [
+                        single_file_exec_path,
+                        "https://www.vistopia.com.cn/article/"
+                        + article["article_id"],
+                        str(fname),
+                        "--browser-cookies-file=" + cookie_file_path
+                    ]
+                    logger.debug(f"singlefile command {command}")
+                    try:
+                        subprocess.run(command, check=True)
+                        print(
+                            f"Successfully fetched and saved to {fname}")
+                    except subprocess.CalledProcessError as e:
+                        print(f"Failed to fetch page using single-file: {e}")
+
     @staticmethod
     def retag(fname, article_info, catalog_info, series_info):
 
         from mutagen.easyid3 import EasyID3
-
-        track = EasyID3(fname)
-        track["title"] = article_info["title"]
-        track["album"] = series_info["title"]
-        track["artist"] = series_info["author"]
-        track["tracknumber"] = article_info["sort_number"]
-        # track["tracksort"] = article_info["sort_number"]
-        track["website"] = article_info["content_url"]
-        track.save()
+        from mutagen.id3._util import ID3NoHeaderError
```

Comment:
The `ID3NoHeaderError` can be imported from `mutagen.id3` directly:
* https://github.com/quodlibet/mutagen/blob/f95d3ae19e25e3f0a91061566551843d799317c5/mutagen/id3/__init__.py#L37
* https://github.com/quodlibet/mutagen/issues/327